### PR TITLE
Implement `GetStateV2` in the beacon API

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Implements API's `GetStateV2` according to the spec: https://ethereum.github.io/beacon-APIs/#/Debug/getStateV2

### CODE REVIEW

File: beacon-chain/rpc/eth/debug/debug.go [hunk 1]
```diff
@@ -50,12 +53,61 @@ func (ds *Server) GetBeaconStateSSZ(ctx context.Context, req *ethpbv1.StateReque
 	return &ethpbv1.BeaconStateSSZResponse{Data: sszState}, nil
 }
 
-func (ds *Server) GetBeaconStateV2(ctx context.Context, request *ethpbv2.StateRequestV2) (*ethpbv2.BeaconStateResponseV2, error) {
-	panic("implement me")
+// GetBeaconStateV2 returns the full beacon state for a given state ID.
+func (ds *Server) GetBeaconStateV2(ctx context.Context, req *ethpbv2.StateRequestV2) (*ethpbv2.BeaconStateResponseV2, error) {
+	ctx, span := trace.StartSpan(ctx, "beaconv1.GetBeaconStateV2")
```
Comment (terencechain) [inline beacon-chain/rpc/eth/debug/debug.go:58]:
```suggestion
	ctx, span := trace.StartSpan(ctx, "beaconV1.GetBeaconStateV2")
```

Better? or should we eliminate v1 all at once? Seems weird to have a v1 and v2

File: beacon-chain/rpc/eth/debug/debug.go [hunk 2]
```diff
@@ -50,12 +53,61 @@ func (ds *Server) GetBeaconStateSSZ(ctx context.Context, req *ethpbv1.StateReque
...
+	default:
+		return nil, status.Error(codes.Internal, "Unsupported state version")
+	}
 }
 
-func (ds *Server) GetBeaconStateSSZV2(ctx context.Context, request *ethpbv2.StateRequestV2) (*ethpbv2.BeaconStateSSZResponseV2, error) {
-	panic("implement me")
+// GetBeaconStateSSZV2 returns the SSZ-serialized version of the full beacon state object for given state ID.
+func (ds *Server) GetBeaconStateSSZV2(ctx context.Context, req *ethpbv2.StateRequestV2) (*ethpbv2.BeaconStateSSZResponseV2, error) {
+	ctx, span := trace.StartSpan(ctx, "beaconv1.GetBeaconStateSSZV2")
```
Comment (terencechain) [inline beacon-chain/rpc/eth/debug/debug.go:97]: Same feedback here

File: beacon-chain/rpc/eth/debug/debug.go (hunk 1 above)
Comment (rkapka) [reply beacon-chain/rpc/eth/debug/debug.go:58]: we should eliminate server versions

Comment (rkapka) [reply beacon-chain/rpc/eth/debug/debug.go]: and it should be called `debug` not `beacon` here

**Review by terencechain — APPROVED** [review]

**Pull request merged**